File: packages/postpin-python/src/postpin/resources/serviceability.py
from __future__ import annotations

from typing import Optional

from .._http import HTTPClient
from ..types import Serviceability as ServiceabilityResult


class Serviceability:
    def __init__(self, http: HTTPClient) -> None:
        self._http = http

    def check(self, pincode: str, *, timeout: Optional[float] = None) -> ServiceabilityResult:
        """Check whether a pincode is serviceable."""
        data, _ = self._http.request("GET", f"/public/serviceability/{pincode}", timeout=timeout)
        return ServiceabilityResult._from(data)

File: packages/postpin-node/src/resources/serviceability.ts
import type { HttpClient } from "../http.js";
import type { RequestOptions, Serviceability } from "../types.js";

/** Quick "can I ship to this pincode?" lookup. */
export class ServiceabilityResource {
  constructor(private readonly http: HttpClient) {}

  /**
   * Check whether a pincode is serviceable and resolve its city/state.
   *
   * @example
   * const s = await postpin.serviceability.check("781001");
   * if (s.serviceable) console.log(`${s.city}, ${s.state}`);
   */
  async check(pincode: string, options: RequestOptions = {}): Promise<Serviceability> {
    const { data } = await this.http.request<Serviceability>({
      method: "GET",
      path: `/public/serviceability/${encodeURIComponent(pincode)}`,
      signal: options.signal,
    });
    return data;
  }
}
